### PR TITLE
fix(projen): uncapping cdk version

### PR DESCRIPTION
**Pull request by oconpa**
Uncapping CDK version allowing this package to be installed via pip less the version error which suggests the installation failed, when in reality it's detected a mismatched version before proceeding to install the latest package version.

Screenshot 2024-01-02 at 8 21 38 PM

Steps to replicate the error:
1. cdk init app --language python
2. python3 -m venv .venv
3. pip install cdklabs.generative-ai-cdk-constructs
4. pip install -r requirements.txt

Fixes #

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ import {
 const GITHUB_USER = 'awslabs';
 const PUBLICATION_NAMESPACE = 'cdklabs';
 const PROJECT_NAME = 'generative-ai-cdk-constructs';
-const CDK_VERSION: string = '2.111.0';
+const CDK_VERSION: string = '2.117.0';
```
Comment (scottschreckengaust) [inline .projenrc.ts:30]: In the process of testing `2.85.0` which is the lowest for `aws-cdk-lib` without an `npm audit` known vulnerability.

**Review by scottschreckengaust — CHANGES_REQUESTED** [review]
Considering lowering as the goal of unpinning would be to maximize compatibilities.

File: .projenrc.ts (hunk 1 above)
Comment (scottschreckengaust) [reply .projenrc.ts:30]: version `2.84.0` has some security vulnerabilities and `2.85.0` currently doesn't:

```
% export CDK_VERSION=2.84.0; npm install aws-cdk-lib@${CDK_VERSION} && npm list  && npm uninstall aws-cdk-lib && npm list

added 5 packages, and audited 40 packages in 2s

3 packages are looking for funding
  run `npm fund` for details

2 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
temp@ /Users/scottschreckengaust/temp
└── aws-cdk-lib@2.84.0


removed 39 packages, and audited 1 package in 641ms

found 0 vulnerabilities
/Users/scottschreckengaust/temp
└── (empty)

% export CDK_VERSION=2.85.0; npm install aws-cdk-lib@${CDK_VERSION} && npm list  && npm uninstall aws-cdk-lib && npm list

added 5 packages, and audited 40 packages in 1s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
temp@ /Users/scottschreckengaust/temp
└── aws-cdk-lib@2.85.0


removed 39 packages, and audited 1 package in 645ms

found 0 vulnerabilities
/Users/scottschreckengaust/temp
└── (empty)
```

Comment (scottschreckengaust) [reply .projenrc.ts:30]: Lowest version that builds is `2.88.0`.

Comment (scottschreckengaust) [inline .projenrc.ts:30]:
```suggestion
const CDK_VERSION: string = '2.88.0';
```

Comment (oconpa) [reply .projenrc.ts]: Version lowered to 2.88.0

Comment (oconpa) [reply .projenrc.ts]: Version lowered and npx projen ran

**Review by scottschreckengaust — APPROVED** [review]
LGTM

**Review by krokoko — APPROVED** [review]

**Pull request merged**